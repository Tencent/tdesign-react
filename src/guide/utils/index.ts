import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
