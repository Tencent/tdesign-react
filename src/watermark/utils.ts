const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
