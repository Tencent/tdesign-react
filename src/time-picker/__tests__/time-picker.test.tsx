import MockDate from 'mockdate';
import { fireEvent, render, waitFor, vi } from '@test/utils';
import React from 'react';
import TimePicker from '../index';

// 固定时间，当使用 new Date() 时，返回固定时间，防止“当前时间”的副作用影响，导致 snapshot 变更，mockdate 插件见 https://github.com/boblauer/MockDate
MockDate.set('2022-08-27');

// TODO
describe('Timepicker 组件测试', () => {
  it('props.disabled works fine', () => {
    // disabled default value is
    const wrapper1 = render(<TimePicker></TimePicker>);
    const container1 = wrapper1.container.querySelector('.t-time-picker');
    expect(container1.querySelector('.t-is-disabled')).toBeFalsy();
    // disabled = true
    const wrapper2 = render(<TimePicker disabled={true}></TimePicker>);
    const container2 = wrapper2.container.querySelector('.t-time-picker .t-input');
    expect(container2).toHaveClass('t-is-disabled');
    // disabled = false
    const wrapper3 = render(<TimePicker disabled={false}></TimePicker>);
    const container3 = wrapper3.container.querySelector('.t-time-picker');
    expect(container3.querySelector('.t-is-disabled')).toBeFalsy();
  });

  it('trigger panel works fine', async () => {
    const { container } = render(<TimePicker></TimePicker>);
    expect(container.querySelectorAll('input').length).toBe(1);
    fireEvent.click(document.querySelector('input'));
    await waitFor(() => {
      expect(document.querySelector('.t-time-picker__panel')).not.toBeNull();
      expect(document.querySelector('.t-time-picker__panel')).toHaveStyle({
        display: 'block',
      });
      expect(document.querySelectorAll('.t-time-picker__panel-body-scroll').length).toBe(3);
    });
  });

  it('props.defaultValue works fine', async () => {
    const { container } = render(<TimePicker defaultValue="00:10:20"></TimePicker>);
    expect(container.querySelectorAll('input').length).toBe(1);
    expect(container.querySelectorAll('input').item(0)).toHaveValue('00:10:20');
    fireEvent.click(document.querySelector('input'));
    await waitFor(() => {
      const scrollPanels = document.querySelectorAll('.t-time-picker__panel-body-scroll');
      expect(scrollPanels.item(0).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('00');
      expect(scrollPanels.item(1).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('10');
      expect(scrollPanels.item(2).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('20');
    });
  });

  it('props.defaultValue for TimePicker works fine', async () => {
    const { container } = render(<TimePicker defaultValue="00:10:20"></TimePicker>);
    expect(container.querySelectorAll('input').length).toBe(1);
    expect(container.querySelectorAll('input').item(0)).toHaveValue('00:10:20');
    fireEvent.click(document.querySelector('input'));
    await waitFor(() => {
      const scrollPanels = document.querySelectorAll('.t-time-picker__panel-body-scroll');
      expect(scrollPanels.item(0).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('00');
      expect(scrollPanels.item(1).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('10');
      expect(scrollPanels.item(2).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('20');
    });
  });

  it('props.defaultValue for TimeRangePicker works fine', async () => {
    const { container } = render(<TimePicker.TimeRangePicker defaultValue={['00:00:00', '00:10:20']} />);
    const inputs = container.querySelectorAll('input');
    expect(inputs.length).toBe(2);
    expect(inputs.item(0)).toHaveValue('00:00:00');
    expect(inputs.item(1)).toHaveValue('00:10:20');
    fireEvent.click(inputs.item(1));
    await waitFor(() => {
      const scrollPanels = document.querySelectorAll('.t-time-picker__panel-body-scroll');
      expect(scrollPanels.item(0).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('00');
      expect(scrollPanels.item(1).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('10');
      expect(scrollPanels.item(2).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('20');
    });
  });

  it('props.value for TimePickerPanel works fine', () => {
    const onChange = vi.fn();
    const { container } = render(<TimePicker.TimePickerPanel value={'00:10:20'} onChange={onChange} />);
    const scrollPanels = container.querySelectorAll('.t-time-picker__panel-body-scroll');
    expect(scrollPanels.item(0).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('00');
    expect(scrollPanels.item(1).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('10');
    expect(scrollPanels.item(2).querySelectorAll('.t-is-current').item(0)).toHaveTextContent('20');
  });

  it('props.allowInput works fine', async () => {
    const handleBlur = vi.fn();
    const handleInput = vi.fn();
    const handleFocus = vi.fn();
    const { container } = render(
      <TimePicker onBlur={handleBlur} onFocus={handleFocus} onInput={handleInput} allowInput />,
    );
    const InputDom = container.querySelector('.t-input__inner');
    fireEvent.focus(InputDom);
    expect(handleFocus).toBeCalledTimes(1);
    expect(handleInput).toBeCalledTimes(1);
    fireEvent.blur(InputDom);
    expect(handleBlur).toBeCalledTimes(1);
  });

  it('props.onInput&onBlur&onForus works fine', async () => {
    const handleBlur = vi.fn();
    const handleInput = vi.fn();
    const handleFocus = vi.fn();
    const { container } = render(
      <TimePicker.TimeRangePicker allowInput onInput={handleInput} onBlur={handleBlur} onFocus={handleFocus} />,
    );
    const inputs = container.querySelectorAll('input');
    fireEvent.focus(inputs[0]);
    expect(handleFocus).toBeCalledTimes(1);
    fireEvent.change(inputs[0], { target: { value: '00:10:20' } });
    expect(handleInput).toBeCalledTimes(1);
    fireEvent.blur(inputs[0]);
    expect(handleBlur).toBeCalledTimes(1);
  });

  it('click to pick', async () => {
    const handleChange = vi.fn();
    const { container } = render(<TimePicker defaultValue="00:00:00" onChange={handleChange}></TimePicker>);
    fireEvent.click(document.querySelector('input'));
    await waitFor(async () => {
      const confirmBtn = document.querySelectorAll('.t-time-picker__panel button').item(0);
      expect(confirmBtn).toBeInTheDocument();
      const panelItem1 = document.querySelectorAll('.t-time-picker__panel-body-scroll').item(0);
      // await fireEvent.scroll(panelItem1, { target: { scrollY: 24 } });
      fireEvent.click(panelItem1.querySelectorAll('.t-time-picker__panel-body-scroll-item').item(1));
      fireEvent.click(confirmBtn);
      expect(container.querySelectorAll('input').item(0)).toHaveValue('01:00:00');
      expect(handleChange).toHaveBeenCalled(1);
    });
  });
});
