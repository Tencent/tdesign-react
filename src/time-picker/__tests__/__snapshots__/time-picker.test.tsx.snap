// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled.jsx 1`] = `
<DocumentFragment>
  <h3>
    禁用整个选择器
  </h3>
  <br />
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input t-select-input--empty"
    >
      <div
        class="t-input__wrap"
        value=""
      >
        <div
          class="t-input t-is-readonly t-is-disabled t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            disabled=""
            placeholder="请选择时间"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <br />
  <br />
  <h3>
    禁用指定时间
  </h3>
  <br />
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input t-select-input--empty"
    >
      <div
        class="t-input__wrap"
        value=""
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请选择时间"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`hide-clear-button.jsx 1`] = `
<DocumentFragment>
  <h3>
    禁止清空
  </h3>
  <br />
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input"
    >
      <div
        class="t-input__wrap"
        value="12:00:00"
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请输入"
            readonly=""
            type="text"
            value="12:00:00"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <br />
  <br />
  <h3>
    允许清空
  </h3>
  <br />
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input"
    >
      <div
        class="t-input__wrap"
        value="12:00:00"
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请输入"
            readonly=""
            type="text"
            value="12:00:00"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`hm.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input"
    >
      <div
        class="t-input__wrap"
        value="12:00"
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请输入"
            readonly=""
            type="text"
            value="12:00"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`hms.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input"
    >
      <div
        class="t-input__wrap"
        value="12:00:00"
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请输入"
            readonly=""
            type="text"
            value="12:00:00"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`keyboard.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input"
    >
      <div
        class="t-input__wrap"
        value="12:08:00"
      >
        <div
          class="t-input t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请输入"
            type="text"
            value="12:08:00"
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`panel.jsx 1`] = `
<DocumentFragment>
  <div>
    <div
      class="t-time-picker__panel"
    >
      <div
        class="t-time-picker__panel-section-body"
      >
        <div
          class="t-time-picker__panel-body"
        >
          <div
            class="t-time-picker__panel-body-active-mask"
          >
            <div />
            <div />
            <div />
          </div>
          <ul
            class="t-time-picker__panel-body-scroll"
          >
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              00
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              01
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              02
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              03
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              04
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              05
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              06
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              07
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item t-is-current"
            >
              08
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              09
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              10
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              11
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              12
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              13
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              14
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              15
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              16
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              17
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              18
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              19
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              20
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              21
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              22
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              23
            </li>
          </ul>
          <ul
            class="t-time-picker__panel-body-scroll"
          >
            <li
              class="t-time-picker__panel-body-scroll-item t-is-current"
            >
              00
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              01
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              02
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              03
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              04
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              05
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              06
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              07
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              08
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              09
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              10
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              11
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              12
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              13
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              14
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              15
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              16
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              17
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              18
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              19
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              20
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              21
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              22
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              23
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              24
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              25
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              26
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              27
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              28
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              29
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              30
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              31
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              32
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              33
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              34
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              35
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              36
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              37
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              38
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              39
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              40
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              41
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              42
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              43
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              44
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              45
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              46
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              47
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              48
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              49
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              50
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              51
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              52
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              53
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              54
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              55
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              56
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              57
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              58
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              59
            </li>
          </ul>
          <ul
            class="t-time-picker__panel-body-scroll"
          >
            <li
              class="t-time-picker__panel-body-scroll-item t-is-current"
            >
              00
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              01
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              02
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              03
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              04
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              05
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              06
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              07
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              08
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              09
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              10
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              11
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              12
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              13
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              14
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              15
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              16
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              17
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              18
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              19
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              20
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              21
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              22
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              23
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              24
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              25
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              26
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              27
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              28
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              29
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              30
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              31
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              32
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              33
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              34
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              35
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              36
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              37
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              38
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              39
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              40
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              41
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              42
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              43
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              44
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              45
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              46
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              47
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              48
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              49
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              50
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              51
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              52
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              53
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              54
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              55
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              56
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              57
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              58
            </li>
            <li
              class="t-time-picker__panel-body-scroll-item"
            >
              59
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`range.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-range-input-popup"
    >
      <div
        class="t-popup__reference t-range-input-popup"
      >
        <div
          class="t-range-input t-time-picker__group"
          value=""
        >
          <div
            class="t-range-input__inner"
          >
            <div
              class="t-range-input__inner-left t-input__wrap"
            >
              <div
                class="t-input t-size-m t-align-left t-is-default"
              >
                <input
                  class="t-input__inner"
                  placeholder="开始时间"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="t-range-input__inner-separator"
            >
              -
            </div>
            <div
              class="t-range-input__inner-right t-input__wrap"
            >
              <div
                class="t-input t-size-m t-align-left t-is-default"
              >
                <input
                  class="t-input__inner"
                  placeholder="结束时间"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <span
              class="t-range-input__suffix t-range-input__suffix-icon"
            >
              <svg
                class="t-icon t-icon-time"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`show-steps.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input t-select-input--empty"
    >
      <div
        class="t-input__wrap"
        value=""
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请选择时间"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`twelve-hour-meridian.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-time-picker"
  >
    <div
      class="t-popup__reference t-time-picker__group t-select-input t-select-input--empty"
    >
      <div
        class="t-input__wrap"
      >
        <div
          class="t-input t-is-readonly t-size-m t-align-left t-is-default t-input--prefix t-input--suffix"
        >
          <div
            class="t-input__prefix"
          />
          <input
            class="t-input__inner"
            placeholder="请选择时间"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="t-input__suffix t-input__suffix-icon"
          >
            <svg
              class="t-icon t-icon-time"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M7.5 4v4.39L10 11l.7-.7-2.2-2.33V4h-1z"
                fill="currentColor"
                fill-opacity="0.9"
              />
              <path
                d="M15 8A7 7 0 111 8a7 7 0 0114 0zm-1 0A6 6 0 102 8a6 6 0 0012 0z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
